test/files/run/wacky-value-classes.scala

```diff
@@ -0,0 +1,19 @@
+// scala/bug#9240
+final class AnyValNothing(val self: Nothing) extends AnyVal
+final class AnyValNull   (val self: Null   ) extends AnyVal
+// scala/bug#10361
+final class AnyValUnit   (val self: Unit   ) extends AnyVal
+
+object Test extends App {
+  def avn = new AnyValNull(null)
+  assert(avn == avn)
+  /*this throws NPE right now b/c null.hashCode() does...*/
+  //assert(avn.hashCode() == 0)
```

Comment:
Sadly, that problem is deeper than this, as far as I can tell. No hat tricks here :disappointed: